src/lib/bootloader/stage1_proposal.rb

```diff
@@ -0,0 +1,256 @@
+require "yast"
+
+Yast.import "Arch"
+Yast.import "BootStorage"
+Yast.import "Storage"
+
+require "bootloader/stage1_device"
+
+module Bootloader
+  # Represents object that can set passed stage1 to proposed values.
+  # It is highly coupled with Stage1 class and it is recommended to use
+  # {Stage1#proposal} instead of direct usage of this class.
+  class Stage1Proposal
+    include Yast::Logger
+
+    # @param [Bootloader::Stage1] stage1 where write proposal
+    def self.propose(stage1)
+      proposal = case Yast::Arch.architecture
```

Comment:
well, when I look at code from longer distance, it is not so bad, it is very visible that this block of code is together. Another option is to use constant same as in grub_install